### ISSUE TITLE
ghc-modi crash

### ISSUE THREAD

**Issue by MarkZander** [atom-haskell/ide-haskell#69]
Haskell-ghc-mod: ghc-modi crashed on type with message NG BUG: GMEIOException /Users/zander/Projects/bassbull/dist/setup-config: hGetContents: invalid argument (invalid byte sequence)
/Users/zander/Projects/bassbull

Even when not crashing I get the following 2 error messages:

Haskell-ghc-mod: ghc-mod -d Main failed with error code 1
/Users/zander/Projects/bassbull/dist/setup-config: hGetContents: invalid argument (invalid byte sequence)

Haskell-ghc-mod: ghc-mod failed with error code 1
/Users/zander/Projects/bassbull/dist/setup-config: hGetContents: invalid argument (invalid byte sequence)

I'm working through the project at http://howistart.org/posts/haskell/1 using ide-haskell and got the above error message.

Atom 0.207.0
ide-haskell 0.5.10
haskell-ghc-mod 0.7.5
ghc-mod 5.2.1.2
I'm on os x 10.10.3, iMac 27-inch late 2013.

tarzander@hotmail.com

**Comment by lierdakil** [comment]
There is a known problem with ghc-mod<=5.2.1.2 and cabal>=1.22. See https://github.com/kazu-yamamoto/ghc-mod/wiki/InconsistentCabalVersions. Workarounds would be to get `master` version of ghc-mod (that's unstable atm) or downgrade cabal to 1.20.x.

**Comment by lierdakil** [comment]
Note that `ghc-mod-master` support is limited atm. See https://github.com/atom-haskell/haskell-ghc-mod/issues/18 for more information

**Comment by MarkZander** [comment]
Thanks for the quick responses. Will try to straighten this out.

Mark

> On Jun 9, 2015, at 3:12 AM, Nikolay Yakimov <notifications@github.com> wrote:
> 
> Note that ghc-mod-master support is limited atm. See atom-haskell/haskell-ghc-mod#18 <https://github.com/atom-haskell/haskell-ghc-mod/issues/18> for more information
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/atom-haskell/ide-haskell/issues/69#issuecomment-110284797>.
> 

On Jun 9, 2015, at 3:10 AM, Nikolay Yakimov <notifications@github.com> wrote:

There is a known problem with ghc-mod<=5.2.1.2 and cabal>=1.22. See https://github.com/kazu-yamamoto/ghc-mod/wiki/InconsistentCabalVersions <https://github.com/kazu-yamamoto/ghc-mod/wiki/InconsistentCabalVersions>. Workarounds would be to get master version of ghc-mod (that's unstable atm) or downgrade cabal to 1.20.x.

**Comment by MarkZander** [comment]
Ok, I followed the procedure as suggested:

$ cabal --version
cabal-install version 1.22.0.0
using version 1.22.0.0 of the Cabal library 
$ cabal install --constraint "Cabal < 1.22" cabal-install
(all kinds of stuff)
Downloading cabal-install-1.18.1.0...
Configuring cabal-install-1.18.1.0...
Building cabal-install-1.18.1.0...
Installed cabal-install-1.18.1.0
$ ghc-pkg list | grep Cabal
    Cabal-1.18.1.5
$ ghc-pkg unregister Cabal-1.22.0.0
ghc-pkg: cannot find package Cabal-1.22.0.0
$ cabal --version
cabal-install version 1.22.0.0
using version 1.22.0.0 of the Cabal library 

So I followed the directions, but it did not replace my binary for cabal which is still being used our of my "~/.cabal/bin" directory. I still get the same errors when using Atom. Any further suggestions?

Thanks, Mark

**Comment by MarkZander** [comment]
Haskell-ghc-mod: ghc-mod -d Main failed with error code 1
/Users/zander/Projects/bassbull/dist/setup-config: hGetContents: invalid argument (invalid byte sequence)

I got the above error and 4 or 5 similar.

After following directions with some amount of trouble:

$ cabal --version
cabal-install version 1.18.1.0
using version 1.18.1.5 of the Cabal library 
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.8.4

Have no idea why the errors persist. Any help is appreciated.

Mark

**Comment by lierdakil** [comment]
If cabal<1.22 is in your PATH, i.e. `cabal --version` gets you 1.18 or 1.20, then you need to run `cabal clean` and `cabal configure` in your project directory, otherwise old files are still there.

**Comment by lierdakil** [comment]
Is this resolved? Or do you still need help with this?

**Comment by MarkZander** [comment]
Thanks, that’s a very generous offer.

It is resolved. I had to be quite thorough in removing everything that I did with cabal 1.22. It’s been quite a struggle to get a reasonable haskell environment up and running. Makes me better appreciate the advantages of the haskell platform.

Mark

**Comment by lierdakil** [comment]
Great.

**Comment by kRITZCREEK** [comment]
@MarkZander You might want to take a look at FPComplete's [Stack](https://www.fpcomplete.com/blog/2015/06/announcing-first-public-beta-stack), which looks to solve the environment problem.

**Comment by MarkZander** [comment]
Thanks for the advice. I’ve got everything working now so I’ll probably stick with what I’ve got for a while.

I looked at stack and I decided against it for now. It claims to do just what I need, whatever that is. Not much on what it is doing and how to get things going like I want. Stack may do just what I need but I guess I’d like to know a little more about what it is doing before I give it a try. Maybe I should write to the stack team and let them know what my situation is.

Thanks, Mark
